python_bindings/src/PyFunc.cpp

```diff
@@ -106,256 +101,214 @@ void define_func(py::module &m) {
-        }, py::arg("dst"), py::arg("target") = Target(), py::arg("param_map") = ParamMap())
-
-        // This will actually allow a list-of-buffers as well as a tuple-of-buffers, but that's OK.
-        .def("realize", [](Func &f, std::vector<Buffer<>> buffers, const Target &t, const ParamMap &param_map) -> void {
-            f.realize(Realization(buffers), t);
-        }, py::arg("dst"), py::arg("target") = Target(), py::arg("param_map") = ParamMap())
-
-        .def("realize", [](Func &f, std::vector<int32_t> sizes, const Target &target, const ParamMap &param_map) -> py::object {
-            return realization_to_object(f.realize(sizes, target, param_map));
-        }, py::arg("sizes") = std::vector<int32_t>{}, py::arg("target") = Target(), py::arg("param_map") = ParamMap())
-
-        // TODO: deprecate in favor of std::vector<int32_t> size version?
-        .def("realize", [](Func &f, int x_size, const Target &target, const ParamMap &param_map) -> py::object {
-            return realization_to_object(f.realize(x_size, target, param_map));
-        }, py::arg("x_size"), py::arg("target") = Target(), py::arg("param_map") = ParamMap())
-
-        // TODO: deprecate in favor of std::vector<int32_t> size version?
-        .def("realize", [](Func &f, int x_size, int y_size, const Target &target, const ParamMap &param_map) -> py::object {
-            return realization_to_object(f.realize(x_size, y_size, target, param_map));
-        }, py::arg("x_size"), py::arg("y_size"), py::arg("target") = Target(), py::arg("param_map") = ParamMap())
-
-        // TODO: deprecate in favor of std::vector<int32_t> size version?
-        .def("realize", [](Func &f, int x_size, int y_size, int z_size, const Target &target, const ParamMap &param_map) -> py::object {
-            return realization_to_object(f.realize(x_size, y_size, z_size, target, param_map));
-        }, py::arg("x_size"), py::arg("y_size"), py::arg("z_size"), py::arg("target") = Target(), py::arg("param_map") = ParamMap())
-
-        // TODO: deprecate in favor of std::vector<int32_t> size version?
-        .def("realize", [](Func &f, int x_size, int y_size, int z_size, int w_size, const Target &target, const ParamMap &param_map) -> py::object {
-            return realization_to_object(f.realize(x_size, y_size, z_size, w_size, target, param_map));
-        }, py::arg("x_size"), py::arg("y_size"), py::arg("z_size"), py::arg("w_size"), py::arg("target") = Target(), py::arg("param_map") = ParamMap())
-
-        .def("defined", &Func::defined)
-        .def("name", &Func::name)
-        .def("dimensions", &Func::dimensions)
-        .def("args", &Func::args)
-        .def("value", &Func::value)
-        .def("values", [](Func &func) -> py::tuple {
-            return to_python_tuple(func.values());
-        })
-        .def("defined", &Func::defined)
-        .def("outputs", &Func::outputs)
-        .def("output_types", &Func::output_types)
+                          .def(py::init<>())
```

Comment:
Perhaps we should uniformly break after the = in these definitions to avoid the massive indent